packages/icon-button/src/IconButton.tsx

```diff
@@ -240,6 +241,17 @@ function usesLinkElement(
   return props.href != null;
 }
 
+function isComponentGlyph(
+  child: React.ReactNode,
+): child is React.ReactElement<LGGlyph.ComponentProps> {
+  return (
+    child != null &&
+    typeof child === 'object' &&
+    'type' in child &&
+    (child.type as any).displayName?.slice(0, 5) === 'Glyph'
```

Comment:
This is pretty gross...I wonder if we just inject a prop on the function so it's more discrete. `Glyph.isGlyph = true` and that lets this check be less fuzzy.

Reply:
👍